syntax = "proto3";

package auth;

message SignupEvent {
  string user_id = 1;
  string email = 2;
  string full_name = 3;
  string role = 4;
}

message LoginEvent {
  string user_id = 1;
  string email = 2;
  string token = 3;
}

service AuthEventEmitter {
  rpc EmitSignup(SignupEvent) returns (EventAck);
  rpc EmitLogin(LoginEvent) returns (EventAck);
}

message EventAck {
  bool success = 1;
  string message = 2;
}
